spec/unit/provider/package/gem_spec.rb

```diff
@@ -185,6 +185,71 @@
         end
       end
     end
+
+    describe 'insync?' do
+      describe 'for array of versions' do
+        let(:is) { ['1.3.4', '3.6.1', '5.1.2'] }
+
+        it 'returns true for ~> 1.3' do
+          resource[:ensure] = '~> 1.3'
+          expect(provider.insync?(is)).to eq(true)
```

Comment:
Is there a reason we shouldn't fix provider.inspect to also output the same information?  I can fix the spec and address the `inspect` issue in another commit.